internal/cluster/phase.go

```diff
@@ -17,38 +17,66 @@ import (
 )
 
 const (
-	OperationPhasePreFlight      = "preflight"
-	OperationPhaseInfrastructure = "infrastructure"
-	OperationPhaseKubernetes     = "kubernetes"
-	OperationPhaseDistribution   = "distribution"
-	OperationPhasePlugins        = "plugins"
-	OperationPhasePreUpgrade     = "preupgrade"
-	OperationPhaseAll            = ""
+	OperationPhasePreFlight             = "preflight"
+	OperationPhaseInfrastructure        = "infrastructure"
+	OperationSubPhasePreInfrastructure  = "pre-infrastructure"
+	OperationSubPhasePostInfrastructure = "post-infrastructure"
+	OperationPhaseKubernetes            = "kubernetes"
+	OperationSubPhasePreKubernetes      = "pre-kubernetes"
+	OperationSubPhasePostKubernetes     = "post-kubernetes"
+	OperationPhaseDistribution          = "distribution"
+	OperationSubPhasePreDistribution    = "pre-distribution"
+	OperationSubPhasePostDistribution   = "post-distribution"
+	OperationPhasePlugins               = "plugins"
+	OperationPhasePreUpgrade            = "pre-upgrade"
+	OperationPhaseAll                   = ""
 
 	OperationPhaseOptionVPNAutoConnect = "vpnautoconnect"
 )
 
-var errUnsupportedPhase = errors.New(
-	"unsupported phase, options are: infrastructure, kubernetes, distribution, plugins",
+var (
+	errUnsupportedPhase = errors.New(
+		"unsupported phase, options are: infrastructure, kubernetes, distribution, plugins",
+	)
+	errUnsupportedOperationPhase = errors.New(
+		"unsupported operation phase, options are: pre-infrastructure, infrastructure, post-infrastructure, " +
+			"pre-kubernetes, kubernetes, post-kubernetes, pre-distribution, distribution, post-distribution, plugins",
+	)
```

Comment:
These errors are returned by exported functions (hence they are visible from other packages) and therefore participate in defining the package's api: this means it would be probably better to export them, too, to let client code handle them should the need arise.